city_scrapers/spiders/cook_north_shore_mosquito.py

```diff
@@ -0,0 +1,98 @@
+import re
+from datetime import datetime
+
+from city_scrapers_core.constants import BOARD
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class CookNorthShoreMosquitoSpider(CityScrapersSpider):
+    name = "cook_north_shore_mosquito"
+    agency = "Cook County North Shore Mosquito Abatement District"
+    timezone = "America/Chicago"
+    allowed_domains = ["www.nsmad.com"]
+    start_urls = ["https://www.nsmad.com/news-events/board-meetings/"]
+    TAG_RE = re.compile(r'<[^>]+>')
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        calendar = response.xpath('//*[@id="post-68"]/div/ul/li/ul')
+        for item in calendar.extract()[0].split('<li>')[1:]:
+            meeting = Meeting(
+                title=self._parse_title(),
+                description='',
+                classification=self._parse_classification(),
+                start=self._parse_start(item, response),
+                all_day=False,
+                time_notes=self._parse_time_notes(item),
+                location=self._parse_location(item),
+                links=self._parse_links(item),
+                source=self._parse_source(response),
+            )
+
+            meeting["status"] = self._get_status(meeting)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_title(self):
+        """Parse or generate meeting title."""
+        return "Board of Trustees"
+
+    def _parse_classification(self):
+        """Parse or generate classification from allowed options."""
+        return BOARD
+
+    def _parse_start(self, item, response):
+        """Parse start datetime as a naive datetime object."""
+        year = '2019'
+        calendar_title = response.xpath('//*[@id="post-68"]/div/h2[1]').extract()[0]
+        if '2019' not in calendar_title:
+            raise ValueError("Year no longer {}".format(year))
+        month_day = self._clean_html_tags(item.split('–')[0])
+        date = month_day + ' ' + year + ' ' + '7 pm'
+        return datetime.strptime(date, '%B %d %Y %I %p')
+
+    def _parse_end(self, item):
+        """Parse end datetime as a naive datetime object. Added by pipeline if None"""
+        return None
+
+    def _parse_time_notes(self, item):
+        """Parse any additional notes on the timing of the meeting"""
```

Comment:
It looks like this is pulling whether the meeting is cancelled, but that should probably just be used for the status instead. It doesn't look like there are notes on the time here otherwise

Reply:
How do I use this for status instead? I've got the check modified and the right constants imported but how can I only sometimes overwrite what `_get_status` returns?

Reply:
It's not super clear, but you can pass an optional `text` keyword argument to `_get_status` that checks if any form of "cancel" or "rescheduled" is in the string and sets it to `CANCELLED` if there is ([here it is in `city-scrapers-core`](https://github.com/City-Bureau/city-scrapers-core/blob/master/city_scrapers_core/spiders/spider.py#L45)).

Here's an example of it in [`chi_community_development`](https://github.com/City-Bureau/city-scrapers/blob/master/city_scrapers/spiders/chi_community_development.py#L50), and you can see where "Cancelled" shows up next to the dates being scraped [on the agency page](https://web.archive.org/web/20190512050940/https://www.chicago.gov/city/en/depts/dcd/supp_info/community_developmentcommission.html)